.github/workflows/Verify_Register_SVD.yml

```diff
@@ -0,0 +1,573 @@
+name: Verify Register and SVD Files
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  MSDK_DIR: msdk
+  MSDK-INTERNAL_DIR: msdk-internal
+  
+  # Check effected parts.
+  # GLOSSARY:
+  # Chip name will mean the designation number of microcontroller like MAX32670 and MAX78000
+  # Die name will refer to the die type name of micrcontroller like ME15 and AI85
+  # Part will mean the microcontroller product itself, could refer to chip name or die name
+  AFFECTED_PARTS: none
+  
+  # Don't run the workflow if none of the register files changed.
+  DIFF_FILES_FLAG: false
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # The verify job will check the differences with any updated register file.
+  verify:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clean
+        continue-on-error: true
+        run: |
+          # Remove local modifications
+          set +e
+          # Attempt to clean the repo
+          git scorch
+          retval=$?
+          # Remove everything if this fails
+          if [[ $retval -ne 0 ]]
+          then
+            rm -rf *
+          fi
+          set -e
+          
+      # Checks-out MSDK and MSDK-INTERNAL repositories under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout msdk repository
+        uses: actions/checkout@v3
+        with:
+          # Update the submodules below, doing so here will convert ssh to https
+          submodules: false
+          fetch-depth: 0
+          path: ${{ env.MSDK_DIR }}
+      
+      - name: Checkout msdk internal repository
+        uses: actions/checkout@v3
+        with:
+          repository: Analog-Devices-MSDK/msdk-internal
+          fetch-depth: 0
+          token: ${{ secrets.WORKFLOW_TOKEN }}
+          path: ${{ env.MSDK-INTERNAL_DIR }}
+          
+      # Use these next steps to verify if register files were updated.
+      # Easier to read in Actions' console window if separated by steps.
+      - name: Checking if register files were updated.
+        run: |
+          cd ${{ env.MSDK_DIR }}
+          CHANGED_FILES=$(git diff --ignore-submodules --name-only remotes/origin/main './Libraries/CMSIS/Device/Maxim/*/Include/*_regs.h' './Libraries/PeriphDrivers/Source/*/*.svd')
```

Comment:
Will this catch changes to Libraries/PeriphDrivers/Source/SYS/SVD/*.svd?

Reply:
Yes, this catches any changes in Libraries/PeriphDrivers/Source/SYS/SVD/*.svd. The asterisk symbol represented as a directory (in './Libraries/PeriphDrivers/Source/*/*.svd') includes all directories and their subdirectories, so the git diff can find any SVD file change in Libraries/PeriphDrivers/Source, no matter the depth which the SVD file is located.

Test 10 below demonstrates what happens when you change a SYS SVD file.